src/http.rs

```diff
@@ -168,41 +194,29 @@ mod tests {
         )
         .unwrap();
 
-        let request_builder = http_client.client.post("http://example.com");
-        let request_with_headers = http_client.attach_headers(request_builder);
+        let _ = http_client
+            .get_json::<serde_json::Value>("http://example.com", Some(15))
```

Comment:
I decided to make the method that we test to be the one that's actually used in other code. One major change is the async interface that caused HashMap to stop working